### PR TITLE
Experience Cloud Site login page Development from scratch using LWC

### PR DESCRIPTION
**Pull request by ManishKumarGupta8972**
Ticket included : [AVE_BLOG54](https://theavenoir.atlassian.net/browse/AVEBLOG-71)
Issue : Experience Cloud Site login page Development from scratch using LWC
Cause : N/A
fix: N/A

### CODE REVIEW

File: force-app/main/default/classes/LoginController.cls [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+    * Allows users to log in to the current site, then takes them to the  startUrl. If startUrl is
+      not a relative path, it defaults to the site's designated index page.
+    * @param : String username, String password
+    * @return : System.PageReference
+    */
+    @AuraEnabled
+    public static String login(String username, String password) {
+        try{
+            String startUrl = '';
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/LoginController.cls:26]: why do we need this variable?

File: force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 2]
```diff
@@ -231,48 +231,6 @@
...
-    <labels>
-        <fullName>Name</fullName>
-        <language>en_US</language>
-        <protected>true</protected>
-        <shortDescription>Name</shortDescription>
-        <value>Name</value>
-    </labels>
-    <labels>
-        <fullName>No_Contacts</fullName>
-        <language>en_US</language>
```
Comment (AvenoirShiv) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:271]: why you removing this label?

Comment (AvenoirShiv) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml:271]: You cannot remove anyone's custom labels

File: force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    handlePasswordChange(event) {
+        this.password = event.target.value;
+    }
+
+    @wire(getIsUsernamePasswordEnabled)
+    wiredIsUsernamePasswordEnabled({ error, data }) {
+        if (data) {
+            this.isUsernamePasswordEnabled = data;
+        } else if (error) {
+            console.alert(error);
```
Comment (AvenoirShiv) [inline force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js:66]: remove console statement. Use show toast event

File: force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    handleClick(event){
+        if(this.username && this.password){
+            event.preventDefault();
+            login({ username: this.username, password: this.password })
+                .then((result) => {
+                    console.log('result ',result);
+                    window.location.href = result;
+                })
+                .catch((error) => {
+                    console.alert(error);
```
Comment (AvenoirShiv) [inline force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js:79]: same

File: force-app/main/default/classes/LoginController.cls (hunk 1 above)
Comment (ManishKumarGupta8972) [reply force-app/main/default/classes/LoginController.cls]: resolved

File: force-app/main/default/labels/CustomLabels.labels-meta.xml (hunk 2 above)
Comment (ManishKumarGupta8972) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml]: Resolved

File: force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js (hunk 3 above)
Comment (ManishKumarGupta8972) [reply force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js]: Showtoast messages won't work on the Experience site login page so I have used the alert function.

File: force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        LINKEDIN_URL,
+        TWITTER_URL
+    };
+
+    @wire(getIsUsernamePasswordEnabled)
+    wiredIsUsernamePasswordEnabled({ error, data }) {
+        if (data) {
+            this.isUsernamePasswordEnabled = data;
+        } else if (error) {
+            alert(error);
```
Comment (AvenoirShiv) [inline force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js:44]: show toast event

File: force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    handleClick(event){
+        if(this.username && this.password){
+            event.preventDefault();
+            login({ username: this.username, password: this.password })
+                .then((result) => {
+                    if (this.isValidURL(result)) {
+                        window.location.href = result;
+                    }
+                    else {
+                        alert(result);
```
Comment (AvenoirShiv) [inline force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js:81]: show toast event

File: force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+                    }
+                })
+                .catch((error) => {
+                    alert(error);
```
Comment (AvenoirShiv) [inline force-app/main/default/lwc/customLoginFormLWC/customLoginFormLWC.js:85]: show toast event

**Review by AvenoirShiv — APPROVED** [review]
looks fine

**Pull request merged**